libsplinter/src/circuit/routing/mod.rs

```diff
@@ -220,11 +220,21 @@ impl Circuit {
     }
 }
 
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum AuthorizationType {
+    Trust,
+    #[cfg(feature = "challenge-authorization")]
+    Challenge {
+        public_key: Vec<u8>,
+    },
+}
+
 /// The routing table representation of a node
 #[derive(Clone, Debug, PartialEq, Eq)]
 pub struct CircuitNode {
     node_id: String,
     endpoints: Vec<String>,
+    authorization_type: AuthorizationType,
```

Comment:
Conceptually this is part of the identifier and it might be clearer in the code if we crated an identifier struct/trait such that all comparisons/lookups happen against the identifier (having multiple parts).